### DISCUSSION TITLE
How to "chain" handlers?

### DISCUSSION THREAD

**Discussion by FreePhoenix888** [deep-foundation/Discussions#29] [Q&A] [answered]
# Abstract
Let us imagine we have A handler which returns specific result
And we want to be called when this result will be available and it will return result too
And another handler can do the same with the previos result and so on...

# Analogue in typescript code
```ts
handlerA(data).then(result => handlerB(result)).then(result => handlerC(result))
```

# Real-Life example
Let us image we have package that lets us use Github Rest Api. There is a handler that lets us to list all organization repositories. We call it and we want to call handler that lets us to create/update file in repository on every repository and so on...

# Question
Do I have write code that waits until the first `Then` link is created and then I can await it and call handlerB and then wait until `Then` link is created (depending on which link/action triggered handlerB call) and then await it and call handlerC and so on... ?

**Comment by konard** [comment, marked as answer]
Looks like it is possible to link handleB with result's type and handleC to link with result's type that is coming out from handlerB. No need for any awaits.

**Reply by FreePhoenix888** [reply]
Do you mean we have to create a link in handler which type is handled by another handler? Does it mean that builtin PromiseResult will not be used by handlerB and handlerC?

**Reply by konard** [reply]
1. Yes.
2. PromiseResult is not designed to be used by next handler, but you still can load it in next handler if you really need it.

**Reply by FreePhoenix888** [reply]
Is it fine to create a link if type `PromiseResult` and name it something like `HandlerAPromiseResult` and create a link of this type in `handlerA` and handle its insertion by `handlerB`?

**Reply by konard** [reply]
Well, i it technically possible, but you will have to insert the instance of that type manually. deeplinks uses exactly PromiseResult type instance once promise is finished.
I don't think it is recommended.
Deep designed to use regular links as input and output for handlers. PromiseResult is usually only for debug.

**Reply by FreePhoenix888** [reply]
Yes. I understand that I will have to insert that link manually. 
Do you recomment to use `Type` instead of `PromiseResult`?

**Reply by konard** [reply]
I think it is much better to use any regular type, as we do in our packages right now.
